snippets/base/templates/base/includes/snippet_js.html

```diff
@@ -355,19 +358,45 @@
         }
     }
 
-    // Download the user's country using the geolocation service.
-    function downloadUserCountry() {
-        $script(GEO_URL, 'geo');
-        $script.ready('geo', function() {
-            try {
-                gSnippetsMap.set('geoCountry', geoip_country_code());
-                gSnippetsMap.set('geoLastUpdated', new Date());
-            } catch (e) {
-                // Most likely failed to load JS file. Continue on without us,
-                // we'll try again next time.
-            }
-        });
-    }
+    {% if client.channel == 'release' %}
+      function downloadUserCountry() {
+          var GEO_URL = "{{ settings.OLD_GEO_URL }}";
+          $script(GEO_URL, 'geo');
+          $script.ready('geo', function() {
+              try {
+                  gSnippetsMap.set('geoCountry', geoip_country_code());
+                  gSnippetsMap.set('geoLastUpdated', new Date());
+              } catch (e) {
+                  // Most likely failed to load JS file. Continue on without us,
+                  // we'll try again next time.
+              }
+          });
+      }
+    {% else %}
+      // Download the user's country using the geolocation service.
+      // Please do not directly use this code or Snippets key.
+      // Contact MLS team for your own credentials.
+      // https://location.services.mozilla.com/contact
+      function downloadUserCountry() {
+          var GEO_URL = "{{ settings.GEO_URL }}";
+          var request = new XMLHttpRequest();
+          request.onreadystatechange = function() {
+              if (request.readyState == 4 && request.status == 200) {
+                  var country_data = JSON.parse(request.responseText);
+
+                  try {
+                      gSnippetsMap.set('geoCountry', country_data.country_code);
+                      gSnippetsMap.set('geoLastUpdated', new Date());
+                  } catch (e) {
+                      // Most likely failed to load Data file. Continue on without us,
+                      // we'll try again next time.
+                  }
+              }
+          };
+          request.open('GET', GEO_URL, true);
+          request.send();
```

Comment:
Interesting. In this case we need to make changes to the server to allow setting the header. Here's the error I'm getting:
```Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://location.services.mozilla.com/v1/country?key=fff72d56-b040-. (Reason: missing token 'content-type' in CORS header 'Access-Control-Allow-Headers' from CORS preflight channel).```

Setting ```Access-Control-Request-Headers:"content-type"``` in the response headers should do it.

If that's easy to do let's go for it, otherwise we can keep JSON.parse.